.github/workflows/build.yml

```diff
@@ -0,0 +1,79 @@
+name: Build
```

Comment:
We have 2 kinds of artifacts here:
1. Container for the external scaler that needs SemVer
2. Containers to help as part of samples

Personally I would split them into two workflows given the sample containers can be pushed every commit and don't have to follow the SemVer of the external scaler itself

Reply:
The GitHub action builds both scaler and sample images but only pushes the scaler image. Since there is no known issue with having a single pipeline, I am leaving it so until we find a limitation with it in future.

I will be working on your README and code-structuring comments. I am leaning more towards naming the namespace as `Keda.Scalers.CosmosDb` so that we can have namespaces in other scaler repos as `Keda.Scalers.AnotherEventType` in future. Let me know your preference.

Reply:
I don't have must of preference on the naming but if we have other scalers, I'd argue that they have to end up in a new repo since they will have their own versioning scheme which would be confusing imo

Reply:
@tomkerkhove - I still am not clear why we should document that this is purely for change feed. I am not aware of any other way Cosmos DB supports polling for new changes to the Container. All listener applications are supposed to have change feeds setup. Let me know if you find a link or reference that proves this to be incorrect.

Reply:
Because there is more to autoscaling Cosmos DB apps than just changefeed :)

People can autoscale based on SQL queries, for example. You are not required to use changefeed, it is just one way of using Cosmos DB.

Reply:
@tomkerkhove - I think there is misunderstanding on your end. Change feeds are not meant to be auto-scaled. They are mechanism provided by Cosmos DB to track changes to the Cosmos containers.

Reply:
OK I assumed the apps being auto-scaled are always trigger-based. That they would only be interested in new updates to the data, instead of looking for an ability to be auto-scaled based on output of a query that runs on the entire data (not just the changes).

Reply:
Yes, they are just different use-cases (and both are fine) but we just have to be explicit and set the right expectations for end-users.

Reply:
Addressed comments.